zarr/core.py

```diff
@@ -175,6 +182,11 @@ def __init__(
         self._is_view = False
         self._partial_decompress = partial_decompress
         self._write_empty_chunks = write_empty_chunks
+        self._meta_array = meta_array
+        if meta_array is not None:
+            self._meta_array = np.empty_like(meta_array)
```

Comment:
Should this be the same size as the other case?

```suggestion
            self._meta_array = np.empty_like(meta_array, shape=())
```